src/com/esotericsoftware/kryo/GenericsResolver.java

```diff
@@ -0,0 +1,34 @@
+package com.esotericsoftware.kryo;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/** Helper class that resolves a type name variable to a concrete class using the current class serialization stack
+ *
+ * @author Jeroen van Erp <jeroen@hierynomus.com> */
+public class GenericsResolver {
+    private Deque<Generics> stack = new ArrayDeque<Generics>();
+
```

Comment:
Actually Java6. If we want to maintain Java5 compatibility, a `LinkedList` might be best.